internal/services/connection_check.go

```diff
@@ -122,7 +122,7 @@ func (cs *ConnectionService) connectionCheck() {
 		cs.admin = admin
 	}
 
-	if cs.isDynamicScalingEnabled() || cs.brokers == nil {
+	if cs.isDynamicScalingEnabled() || cs.canaryConfig.ExpectedClusterSize != len(cs.brokers) {
```

Comment:
Does `len(nil)` return 0?

Reply:
i had that question, I satisfied myself that if cs.brokers == nil, len(cs.brokers) == 0 https://pkg.go.dev/builtin#len

Reply:
Yeah, in the end a "slice" in Golang is a struct with three fields: len, cap and a pointer. A `nil` slice has a len = cap = 0 and a pointer = nil and for it the `len` function returns the actual len which is 0 in the struct. An empty slice is different because the pointer is not `nil` but pointing to a backing array which can be empty, so in that case the `len` in the slice is still 0 but it's not a `nil` slice but an empty one. In the end, it's the len field of the slice which makes sense for the function.